File: Modules/HumanResource/Announcement/AnnouncementSectionBinded/AnnouncementSectionBindedForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.Announcement.Forms;

[FormScript("Announcement.AnnouncementSectionBinded")]
[BasedOnRow(typeof(AnnouncementSectionBindedRow), CheckNames = true)]
public class AnnouncementSectionBindedForm
{
    public int AnnouncementWizardId { get; set; }
    public int SectionId { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public short IsActive { get; set; }
    public int InsertUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int DeleteUserId { get; set; }
}

File: Modules/ServerTypes/Announcement/AnnouncementSectionBindedForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AnnouncementSectionBindedForm {
    AnnouncementWizardId: IntegerEditor;
    SectionId: IntegerEditor;
    InsertDate: DateEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    IsActive: IntegerEditor;
    InsertUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    DeleteUserId: IntegerEditor;
}

export class AnnouncementSectionBindedForm extends PrefixedContext {
    static formKey = 'Announcement.AnnouncementSectionBinded';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AnnouncementSectionBindedForm.init)  {
            AnnouncementSectionBindedForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(AnnouncementSectionBindedForm, [
                'AnnouncementWizardId', w0,
                'SectionId', w0,
                'InsertDate', w1,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'IsActive', w0,
                'InsertUserId', w0,
                'UpdateUserId', w0,
                'DeleteUserId', w0
            ]);
        }
    }
}